File: python/Can Place Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        ctr = 0
        # l = len(flowerbed)

        if n == 0:
            return True
        elif len(flowerbed) == 1:
            return (n < 2 and flowerbed[0] == 0)

        i = 0

        while i < len(flowerbed):
            if i == 0:
                if (flowerbed[i] or flowerbed[i+1]) == 0:
                    ctr += 1
                    flowerbed[i] = 1
                i += 1
                continue    

            elif i == len(flowerbed)-1:

                if (flowerbed[i-1] or flowerbed[i]) == 0:
                    ctr += 1
                    break

            else:

                # if flowerbed[i-1] == 1:
                #     continue

                if (flowerbed[i-1] or flowerbed[i] or flowerbed[i+1]) == 0:
                    ctr += 1
                    flowerbed[i] = 1
                    i += 1
                    continue

            i += 1

        return ctr >= n

File: C++/Can Place Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        
        int ctr = 0;
        size_t s = flowerbed.size();

        if (n == 0) return true;
        else if (s == 1) return (n < 2 && flowerbed[0] == 0);

        for (size_t i = 0; i < s; i++) {

            if (i == 0) {
                if (!(flowerbed[0] || flowerbed[1])) {
                    ctr++;
                    flowerbed[0] = 1;
                }
                continue;
            } else if (i == s-1) {

                if (!(flowerbed[i-1] || flowerbed[i])) {
                    ctr++;
                    break;
                }

            } else {

                if (!(flowerbed[i-1] || flowerbed[i] || flowerbed[i+1])) {
                    ctr++;
                    flowerbed[i] = 1;
                }
                continue;
            }

        }
        
        return (ctr >= n);

    }
};
